contrib/api_update.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# See file LICENSE for terms.
+#
+
+set -euE
+
+branch="${1?Specify remote version to use like "upstream/v1.18.x"}"
+remote="${branch%%/*}"
+version="$(echo "${branch##*/}" | sed -e 's@\(v[0-9]\+\.[0-9]\+\).*@\1@')"
+
+if ! git rev-parse --verify --quiet "$branch" >/dev/null; then
+    echo "Branch \"$branch\" does not exist"
+    exit 1
+fi
+
+echo "Proceed with clean checkout of $branch (${version})?"
+read -r
+
+set -x
+
+sudo yum -y install doxygen doxygen-latex
+
+git checkout "$branch"
+git clean -xdf
+git reset --hard
```

Comment:
it looks dangerous. Maybe mention it explicitly/(more clear) on line 18

Reply:
added check to address comment